tests/e2e/cucumber/step_definitions/app.files.ts

```diff
@@ -169,6 +169,39 @@ Given(
     }
   }
 )
+When(
+  '{string} downloads the following file(s) using batch action',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorsEnvironment.getActor({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    let files, downloads
+    await allFilesPage.navigate()
+    // Archiver download is not implemented in OC10 yet
+    // https://github.com/owncloud/web/issues/6239
+    // so this step is skipped in OC10, make sure to remove
+    // the check when the feature gets implemented
+    if (config.ocis) {
+      const downloadInfo = stepTable.hashes().reduce((acc, stepRow) => {
+        const { resource, from } = stepRow
+        if (!acc[from]) {
+          acc[from] = []
+        }
```

Comment:
As suggested in https://github.com/owncloud/web/issues/6389#issuecomment-1060580072 I've skipped this is OC10 and put a comment. Need suggestion on if this approach is okay or we want to write separate tests according to the behavior of diff servers. In this case, writing separate tests means not using this step for `OC10` as other steps will be the same, which is almost equivalent to skipping it here. 

cc: @individual-it @pascalwengerter @fschade @kulmann

Reply:
I would not hide those "workarounds" in the test code. With that the step does not do what it promises.
If this feature does not work in oc10 then I think we need to have an extra scenario for oc10 not using this step